docs/hoon/basic.md

```diff
@@ -7,85 +7,125 @@ title: Basic types
 
 # Basic types
 
-A `span` is a set of nouns and an interpretation of these nouns.
-Officially, we don't use the word "type" in Hoon.  But if we had
-to say "type," it would mean `span`.
+A Hoon `type` is a set of nouns. There is no syntax for directly 
+defining the data type of a Hoon expression. Instead, the compiler 
+infers the type of a Hoon expression by calculating the range of 
+possible values of that expression.
+
+Working with a type in Hoon requires the use of a `mold`. Each 
+type is defined by a mold. To define a custom type, you must create 
+a custom mold.
```

Comment:
A think this language is over constrained. While it's certainly the typical pattern, "requires" and "must" aren't justified, IMO.

Reply:
Earnest question, how else may we use types?  I hesitated when writing this, because of `^+`, but that only works because there is a mold for the example that `^+` uses.  So while it's a little indirect, a mold is indeed needed there.